rclcpp/include/rclcpp/memory_strategy.hpp

```diff
@@ -65,8 +90,22 @@ class MemoryStrategy
    */
   virtual void return_handles(HandleType type, void ** handles)
   {
-    (void)type;
-    this->free(handles);
+    switch (type) {
+      case HandleType::subscription_handle:
+        memset(handles, 0, subscription_handles.size());
+        break;
+      case HandleType::service_handle:
+        memset(handles, 0, service_handles.size());
+        break;
+      case HandleType::client_handle:
+        memset(handles, 0, client_handles.size());
+        break;
+      case HandleType::guard_condition_handle:
+        guard_cond_handles.fill(0);
+        break;
+      default:
+        throw std::runtime_error("Unrecognized enum, could not borrow handle memory.");
+    }
```

Comment:
Is it worth checking that the value in handles matches the address of the pointer returned by `.data()` of the corresponding vector?